diff --git a/bfd/config.bfd b/bfd/config.bfd
index 4e5a838fe4e..2c6a35d918b 100644
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -705,6 +705,11 @@ case "${targ}" in
     targ_selvecs="i386_elf32_vec"
     want64=true
     ;;
+  x86_64-*-hades*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
   x86_64-*-netbsd* | x86_64-*-openbsd*)
     targ_defvec=x86_64_elf64_vec
     targ_selvecs="i386_elf32_vec iamcu_elf32_vec i386_coff_vec i386_pei_vec x86_64_pe_vec x86_64_pei_vec"
diff --git a/config.sub b/config.sub
index 4aaae46f6f7..567bac5c612 100755
--- a/config.sub
+++ b/config.sub
@@ -2023,6 +2023,7 @@ case $os in
 	| iris* \
 	| irix* \
 	| ironclad* \
+    | hades* \
 	| isc* \
 	| its* \
 	| l4re* \
diff --git a/gas/configure.tgt b/gas/configure.tgt
index 7c81bffa16b..6f0267cd884 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -231,6 +231,7 @@ case ${generic_target} in
   i386-*-elf*)				fmt=elf ;;
   i386-*-fuchsia*)			fmt=elf ;;
   i386-*-haiku*)			fmt=elf em=haiku ;;
+  i386-*-hades*)			fmt=elf em=gnu ;;
   i386-*-genode*)			fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-netbsd*-gnu* | \
diff --git a/ld/configure.tgt b/ld/configure.tgt
index ea01ccf9a1b..ee68f6533b1 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -1005,6 +1005,9 @@ x86_64-*-cloudabi*)	targ_emul=elf_x86_64_cloudabi
 x86_64-*-haiku*)	targ_emul=elf_x86_64_haiku
 			targ_extra_emuls="elf_x86_64 elf_i386_haiku"
 			;;
+x86_64-*-hades*)	targ_emul=elf_x86_64
+			targ_extra_emuls="elf_i386"
+			;;
 x86_64-*-linux-gnux32)	targ_emul=elf32_x86_64
 			targ_extra_emuls="elf_x86_64 elf_i386 elf_iamcu"
 			targ_extra_libpath="elf_i386 elf_iamcu elf_x86_64"
