diff -Naur ../xz-utils-v5.4.4-orig/build-aux/config.guess ./build-aux/config.guess
--- ../xz-utils-v5.4.4-orig/build-aux/config.guess      2024-12-28 02:04:37.166387888 -0500
+++ ./build-aux/config.guess    2024-12-28 02:04:40.942422036 -0500
@@ -963,6 +963,12 @@
        GNU_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
        GUESS=$UNAME_MACHINE-unknown-$GNU_SYS$GNU_REL-$LIBC
        ;;
+    x86_64:[Hh]ades:*:*|i?86:[Hh]ades:*:*)
+    GUESS="$UNAME_MACHINE-pc-hades"
+    ;;
+    *:[Hh]ades:*:*)
+    GUESS="$UNAME_MACHINE-unknown-hades"
+    ;;    
     *:Minix:*:*)
        GUESS=$UNAME_MACHINE-unknown-minix
        ;;
diff -Naur ../xz-utils-v5.4.4-orig/build-aux/config.sub ./build-aux/config.sub
--- ../xz-utils-v5.4.4-orig/build-aux/config.sub        2024-12-28 02:04:37.166387888 -0500
+++ ./build-aux/config.sub      2024-12-28 02:04:44.134450900 -0500
@@ -1723,7 +1723,7 @@
             | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
             | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
             | hiux* | abug | nacl* | netware* | windows* \
-            | os9* | macos* | osx* | ios* \
+            | os9* | macos* | osx* | ios* | hades* \
             | mpw* | magic* | mmixware* | mon960* | lnews* \
             | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
             | aos* | aros* | cloudabi* | sortix* | twizzler* \