diff --git a/data/ostable b/data/ostable
index 64f424490..39ea16eba 100644
--- a/data/ostable
+++ b/data/ostable
@@ -39,3 +39,4 @@ base-bsd-openbsd	openbsd			openbsd[^-]*
 base-sysv-aix		aix			aix[^-]*
 base-sysv-solaris	solaris			solaris[^-]*
 base-tos-mint		mint			mint[^-]*
+base-pc-hades          hades                   hades[^-]*
\ No newline at end of file
diff --git a/data/tupletable b/data/tupletable
index ae9f2ddb4..9d1f12d27 100644
--- a/data/tupletable
+++ b/data/tupletable
@@ -67,3 +67,4 @@ base-sysv-solaris-i386		solaris-i386
 base-sysv-solaris-sparc		solaris-sparc
 base-sysv-solaris-sparc64	solaris-sparc64
 base-tos-mint-m68k		mint-m68k
+base-pc-hades-amd64            hades-amd64
\ No newline at end of file
diff --git a/lib/dpkg/arch.c b/lib/dpkg/arch.c
index 9409f5a5e..668926712 100644
--- a/lib/dpkg/arch.c
+++ b/lib/dpkg/arch.c
@@ -286,7 +286,7 @@ dpkg_arch_load_list(void)
 {
 	FILE *fp;
 	char *archfile;
-	char archname[_POSIX2_LINE_MAX];
+	char archname[2048];
 
 	archfile = dpkg_db_get_path(DPKG_DB_ARCH_FILE);
 	fp = fopen(archfile, "r");
diff --git a/lib/dpkg/progname.c b/lib/dpkg/progname.c
index 65592cc25..0f19912f9 100644
--- a/lib/dpkg/progname.c
+++ b/lib/dpkg/progname.c
@@ -50,7 +50,7 @@ dpkg_set_progname(const char *name)
 }
 
 #if defined(HAVE___PROGNAME)
-extern const char *__progname;
+extern char *__progname;
 #endif
 
 /**
diff --git a/lib/dpkg/trigdeferred.c b/lib/dpkg/trigdeferred.c
index ae31d6285..fa28c011a 100644
--- a/lib/dpkg/trigdeferred.c
+++ b/lib/dpkg/trigdeferred.c
@@ -182,7 +182,7 @@ trigdef_skip_whitespace(char *ptr)
 int
 trigdef_parse(void)
 {
-	char line[_POSIX2_LINE_MAX];
+	char line[2048];
 	char *ptr, *ptr_ini;
 	int line_num = 0;
 
