diff -Naur ../bash-5.2-orig/builtins/psize.c ./builtins/psize.c
--- ../bash-5.2-orig/builtins/psize.c	2024-12-24 00:06:55.883169866 -0500
+++ ./builtins/psize.c	2024-12-24 00:12:24.367547363 -0500
@@ -21,33 +21,14 @@
 /*  Write output in 128-byte chunks until we get a sigpipe or write gets an
     EPIPE.  Then report how many bytes we wrote.  We assume that this is the
     pipe size. */
-#include <config.h>
-
-#if defined (HAVE_UNISTD_H)
-#  ifdef _MINIX
-#    include <sys/types.h>
-#  endif
-#  include <unistd.h>
-#endif
-
-#include <stdio.h>
-#ifndef _MINIX
-#include "../bashtypes.h"
-#endif
 #include <signal.h>
-#include <errno.h>
-
-#include "../command.h"
-#include "../general.h"
-#include "../sig.h"
-
-#ifndef errno
-extern int errno;
-#endif
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
 
 int nw;
 
-sighandler
+void
 sigpipe (sig)
      int sig;
 {
diff -Naur ../bash-5.2-orig/lib/termcap/termcap.c ./lib/termcap/termcap.c
--- ../bash-5.2-orig/lib/termcap/termcap.c	2024-12-24 00:06:55.819169818 -0500
+++ ./lib/termcap/termcap.c	2024-12-24 00:20:58.826387651 -0500
@@ -19,6 +19,7 @@
 */
 
 /* Emacs config.h may rename various library functions such as malloc.  */
+#include <stdio.h>
 #ifdef HAVE_CONFIG_H
 
 #include <config.h>
@@ -630,7 +631,7 @@
   bufp->ateof = 0;
   *bufp->ptr = '\0';
 
-  lseek (fd, 0L, 0);
+  lseek (fd, 0L, SEEK_SET);
 
   while (!bufp->ateof)
     {
diff -Naur ../bash-v5.2/lib/readline/terminal.c ./lib/readline/terminal.c
--- ../bash-v5.2/lib/readline/terminal.c        2022-04-05 10:44:17.000000000 -0400
+++ ./lib/readline/terminal.c   2024-12-24 18:22:30.341532747 -0500
@@ -101,13 +101,7 @@
 #endif
 
 static int tcap_initialized;
-
-#if !defined (__linux__) && !defined (NCURSES_VERSION)
-#  if defined (__EMX__) || defined (NEED_EXTERN_PC)
-extern 
-#  endif /* __EMX__ || NEED_EXTERN_PC */
-char PC, *BC, *UP;
-#endif /* !__linux__ && !NCURSES_VERSION */
+extern char PC, *BC, *UP;
 
 /* Some strings to control terminal actions.  These are output by tputs (). */
 char *_rl_term_clreol;
diff -Naur ../bash-5.2-orig/mksyntax.c ./mksyntax.c
--- ../bash-5.2-orig/mksyntax.c	2024-12-24 00:06:55.867169854 -0500
+++ ./mksyntax.c	2024-12-24 00:13:44.532306206 -0500
@@ -20,30 +20,19 @@
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "config.h"
-
 #include <stdio.h>
+#include <string.h>
 #include "bashansi.h"
 #include "chartypes.h"
 #include <errno.h>
 
-#ifdef HAVE_UNISTD_H
-#  include <unistd.h>
-#endif
+#include <unistd.h>
 
 #include "syntax.h"
 
 extern int optind;
 extern char *optarg;
 
-#ifndef errno
-extern int errno;
-#endif
-
-#ifndef HAVE_STRERROR
-extern char *strerror();
-#endif
-
 struct wordflag {
 	int	flag;
 	char	*fstr;
@@ -374,42 +363,4 @@
   if (fp != stdout)
     fclose (fp);
   exit (0);
-}
-
-
-#if !defined (HAVE_STRERROR)
-
-#include <bashtypes.h>
-#if defined (HAVE_SYS_PARAM_H)
-#  include <sys/param.h>
-#endif
-
-#if defined (HAVE_UNISTD_H)
-#  include <unistd.h>
-#endif
-
-/* Return a string corresponding to the error number E.  From
-   the ANSI C spec. */
-#if defined (strerror)
-#  undef strerror
-#endif
-
-char *
-strerror (e)
-     int e;
-{
-  static char emsg[40];
-#if defined (HAVE_SYS_ERRLIST)
-  extern int sys_nerr;
-  extern char *sys_errlist[];
-
-  if (e > 0 && e < sys_nerr)
-    return (sys_errlist[e]);
-  else
-#endif /* HAVE_SYS_ERRLIST */
-    {
-      sprintf (emsg, "Unknown system error %d", e);
-      return (&emsg[0]);
-    }
-}
-#endif /* HAVE_STRERROR */
+}
\ No newline at end of file
diff -Naur ../bash-5.2-orig/shell.c ./shell.c
--- ../bash-5.2-orig/shell.c	2024-12-24 00:06:55.855169845 -0500
+++ ./shell.c	2024-12-24 00:21:36.514377318 -0500
@@ -1647,7 +1647,7 @@
 #endif
 
   /* Only do this with non-tty file descriptors we can seek on. */
-  if (fd_is_tty == 0 && (lseek (fd, 0L, 1) != -1))
+  if (fd_is_tty == 0 && (lseek (fd, 0L, SEEK_CUR) != -1))
     {
       /* Check to see if the `file' in `bash file' is a binary file
 	 according to the same tests done by execute_simple_command (),
@@ -1684,7 +1684,7 @@
 	  exit (EX_BINARY_FILE);
 	}
       /* Now rewind the file back to the beginning. */
-      lseek (fd, 0L, 0);
+      lseek (fd, 0L, SEEK_SET);
     }
 
   /* Open the script.  But try to move the file descriptor to a randomly
diff -Naur ../bash-5.2-orig/support/bashversion.c ./support/bashversion.c
--- ../bash-5.2-orig/support/bashversion.c	2024-12-24 00:06:55.819169818 -0500
+++ ./support/bashversion.c	2024-12-24 00:14:27.644114325 -0500
@@ -17,16 +17,12 @@
    You should have received a copy of the GNU General Public License
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
-
-#include "config.h"
-
 #include "stdc.h"
 
 #include <stdio.h>
+#include <string.h>
 
-#if defined (HAVE_UNISTD_H)
-#  include <unistd.h>
-#endif
+#include <unistd.h>
 
 #include "bashansi.h"
 
diff -Naur ../bash-5.2-orig/support/config.sub ./support/config.sub
--- ../bash-5.2-orig/support/config.sub	2024-12-24 00:06:55.819169818 -0500
+++ ./support/config.sub	2024-12-24 00:20:17.422402229 -0500
@@ -1728,7 +1728,7 @@
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
 	     | hiux* | abug | nacl* | netware* | windows* \
-	     | os9* | macos* | osx* | ios* \
+	     | os9* | macos* | osx* | ios* | hades* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
 	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
diff -Naur ../bash-5.2-orig/support/man2html.c ./support/man2html.c
--- ../bash-5.2-orig/support/man2html.c	2024-12-24 00:06:55.819169818 -0500
+++ ./support/man2html.c	2024-12-24 00:15:11.055276557 -0500
@@ -62,10 +62,6 @@
  *    that all these features work on all manpages. (I didn't have the
  *    time to look through all the available manpages.)
  */
-#ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
-
 #define NROFF 0
 
 #include <unistd.h>
@@ -123,27 +119,6 @@
 #define BD_LITERAL  1
 #define BD_INDENT   2
 
-#ifndef HAVE_STRERROR
-static char *
-strerror(int e)
-{
-	static char emsg[40];
-
-#if defined (HAVE_SYS_ERRLIST)
-	extern int sys_nerr;
-	extern char *sys_errlist[];
-
-	if (e > 0 && e < sys_nerr)
-		return (sys_errlist[e]);
-	else
-#endif				/* HAVE_SYS_ERRLIST */
-	{
-		sprintf(emsg, "Unknown system error %d", e);
-		return (&emsg[0]);
-	}
-}
-#endif				/* !HAVE_STRERROR */
-
 static char *
 strgrow(char *old, int len)
 {
diff -Naur ../bash-5.2-orig/support/mksignames.c ./support/mksignames.c
--- ../bash-5.2-orig/support/mksignames.c	2024-12-24 00:06:55.819169818 -0500
+++ ./support/mksignames.c	2024-12-24 00:16:05.286787255 -0500
@@ -18,18 +18,11 @@
    You should have received a copy of the GNU General Public License
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
-
-#include <config.h>
-
 #include <sys/types.h>
 #include <signal.h>
 
 #include <stdio.h>
-#if defined (HAVE_STDLIB_H)
-#  include <stdlib.h>
-#else
-#  include "ansi_stdlib.h"
-#endif /* HAVE_STDLIB_H */
+#include <stdlib.h>
 
 /* Duplicated from signames.c */
 #if !defined (NSIG)
diff -Naur ../bash-5.2-orig/support/signames.c ./support/signames.c
--- ../bash-5.2-orig/support/signames.c	2024-12-24 00:06:55.819169818 -0500
+++ ./support/signames.c	2024-12-24 00:16:32.922659755 -0500
@@ -17,19 +17,12 @@
    You should have received a copy of the GNU General Public License
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
-
-#include <config.h>
-
 #include <stdio.h>
 
 #include <sys/types.h>
 #include <signal.h>
 
-#if defined (HAVE_STDLIB_H)
-#  include <stdlib.h>
-#else
-#  include "ansi_stdlib.h"
-#endif /* HAVE_STDLIB_H */
+#include <stdlib.h>
 
 #if !defined (NSIG)
 #  define NSIG 64
