diff -Naur --exclude .git ../bash-v5.1-orig/builtins/psize.c ./builtins/psize.c
--- ../bash-v5.1-orig/builtins/psize.c	2025-01-02 18:28:47.206904581 -0500
+++ ./builtins/psize.c	2025-01-02 18:33:28.045245420 -0500
@@ -21,33 +21,14 @@
 /*  Write output in 128-byte chunks until we get a sigpipe or write gets an
     EPIPE.  Then report how many bytes we wrote.  We assume that this is the
     pipe size. */
-#include <config.h>
-
-#if defined (HAVE_UNISTD_H)
-#  ifdef _MINIX
-#    include <sys/types.h>
-#  endif
-#  include <unistd.h>
-#endif
-
-#include <stdio.h>
-#ifndef _MINIX
-#include "../bashtypes.h"
-#endif
 #include <signal.h>
-#include <errno.h>
-
-#include "../command.h"
-#include "../general.h"
-#include "../sig.h"
-
-#ifndef errno
-extern int errno;
-#endif
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
 
 int nw;
 
-sighandler
+void
 sigpipe (sig)
      int sig;
 {
diff -Naur --exclude .git ../bash-v5.1-orig/lib/termcap/termcap.c ./lib/termcap/termcap.c
--- ../bash-v5.1-orig/lib/termcap/termcap.c	2025-01-02 18:28:47.230904343 -0500
+++ ./lib/termcap/termcap.c	2025-01-02 18:37:19.493792593 -0500
@@ -627,7 +627,7 @@
   bufp->ateof = 0;
   *bufp->ptr = '\0';
 
-  lseek (fd, 0L, 0);
+  lseek (fd, 0L, SEEK_SET);
 
   while (!bufp->ateof)
     {
diff -Naur --exclude .git ../bash-v5.1-orig/mksyntax.c ./mksyntax.c
--- ../bash-v5.1-orig/mksyntax.c	2025-01-02 18:28:47.198904660 -0500
+++ ./mksyntax.c	2025-01-02 18:34:43.073546982 -0500
@@ -19,31 +19,19 @@
    You should have received a copy of the GNU General Public License
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
-
-#include "config.h"
-
 #include <stdio.h>
+#include <string.h>
 #include "bashansi.h"
 #include "chartypes.h"
 #include <errno.h>
 
-#ifdef HAVE_UNISTD_H
-#  include <unistd.h>
-#endif
+#include <unistd.h>
 
 #include "syntax.h"
 
 extern int optind;
 extern char *optarg;
 
-#ifndef errno
-extern int errno;
-#endif
-
-#ifndef HAVE_STRERROR
-extern char *strerror();
-#endif
-
 struct wordflag {
 	int	flag;
 	char	*fstr;
@@ -374,42 +362,4 @@
   if (fp != stdout)
     fclose (fp);
   exit (0);
-}
-
-
-#if !defined (HAVE_STRERROR)
-
-#include <bashtypes.h>
-#if defined (HAVE_SYS_PARAM_H)
-#  include <sys/param.h>
-#endif
-
-#if defined (HAVE_UNISTD_H)
-#  include <unistd.h>
-#endif
-
-/* Return a string corresponding to the error number E.  From
-   the ANSI C spec. */
-#if defined (strerror)
-#  undef strerror
-#endif
-
-char *
-strerror (e)
-     int e;
-{
-  static char emsg[40];
-#if defined (HAVE_SYS_ERRLIST)
-  extern int sys_nerr;
-  extern char *sys_errlist[];
-
-  if (e > 0 && e < sys_nerr)
-    return (sys_errlist[e]);
-  else
-#endif /* HAVE_SYS_ERRLIST */
-    {
-      sprintf (emsg, "Unknown system error %d", e);
-      return (&emsg[0]);
-    }
-}
-#endif /* HAVE_STRERROR */
+}
\ No newline at end of file
diff -Naur --exclude .git ../bash-v5.1-orig/shell.c ./shell.c
--- ../bash-v5.1-orig/shell.c	2025-01-02 18:28:47.202904621 -0500
+++ ./shell.c	2025-01-02 18:38:00.585785272 -0500
@@ -1614,7 +1614,7 @@
 #endif
 
   /* Only do this with non-tty file descriptors we can seek on. */
-  if (fd_is_tty == 0 && (lseek (fd, 0L, 1) != -1))
+  if (fd_is_tty == 0 && (lseek (fd, 0L, SEEK_CUR) != -1))
     {
       /* Check to see if the `file' in `bash file' is a binary file
 	 according to the same tests done by execute_simple_command (),
@@ -1651,7 +1651,7 @@
 	  exit (EX_BINARY_FILE);
 	}
       /* Now rewind the file back to the beginning. */
-      lseek (fd, 0L, 0);
+      lseek (fd, 0L, SEEK_SET);
     }
 
   /* Open the script.  But try to move the file descriptor to a randomly
diff -Naur --exclude .git ../bash-v5.1-orig/support/bashversion.c ./support/bashversion.c
--- ../bash-v5.1-orig/support/bashversion.c	2025-01-02 18:28:47.242904223 -0500
+++ ./support/bashversion.c	2025-01-02 18:35:17.073641301 -0500
@@ -18,15 +18,12 @@
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "config.h"
-
 #include "stdc.h"
 
 #include <stdio.h>
+#include <string.h>
 
-#if defined (HAVE_UNISTD_H)
-#  include <unistd.h>
-#endif
+#include <unistd.h>
 
 #include "bashansi.h"
 
diff -Naur --exclude .git ../bash-v5.1-orig/support/config.sub ./support/config.sub
--- ../bash-v5.1-orig/support/config.sub	2025-01-02 18:28:47.242904223 -0500
+++ ./support/config.sub	2025-01-02 18:36:51.693782005 -0500
@@ -1695,7 +1695,7 @@
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
 	     | hiux* | abug | nacl* | netware* | windows* \
-	     | os9* | macos* | osx* | ios* \
+	     | os9* | macos* | osx* | ios* | hades* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
 	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
diff -Naur --exclude .git ../bash-v5.1-orig/support/man2html.c ./support/man2html.c
--- ../bash-v5.1-orig/support/man2html.c	2025-01-02 18:28:47.242904223 -0500
+++ ./support/man2html.c	2025-01-02 18:35:48.197706523 -0500
@@ -62,10 +62,6 @@
  *    that all these features work on all manpages. (I didn't have the
  *    time to look through all the available manpages.)
  */
-#ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
-
 #define NROFF 0
 
 #include <unistd.h>
@@ -123,27 +119,6 @@
 #define BD_LITERAL  1
 #define BD_INDENT   2
 
-#ifndef HAVE_STRERROR
-static char *
-strerror(int e)
-{
-	static char emsg[40];
-
-#if defined (HAVE_SYS_ERRLIST)
-	extern int sys_nerr;
-	extern char *sys_errlist[];
-
-	if (e > 0 && e < sys_nerr)
-		return (sys_errlist[e]);
-	else
-#endif				/* HAVE_SYS_ERRLIST */
-	{
-		sprintf(emsg, "Unknown system error %d", e);
-		return (&emsg[0]);
-	}
-}
-#endif				/* !HAVE_STRERROR */
-
 static char *
 strgrow(char *old, int len)
 {
diff -Naur --exclude .git ../bash-v5.1-orig/support/mksignames.c ./support/mksignames.c
--- ../bash-v5.1-orig/support/mksignames.c	2025-01-02 18:28:47.242904223 -0500
+++ ./support/mksignames.c	2025-01-02 18:36:08.813739219 -0500
@@ -19,17 +19,11 @@
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <config.h>
-
 #include <sys/types.h>
 #include <signal.h>
 
 #include <stdio.h>
-#if defined (HAVE_STDLIB_H)
-#  include <stdlib.h>
-#else
-#  include "ansi_stdlib.h"
-#endif /* HAVE_STDLIB_H */
+#include <stdlib.h>
 
 /* Duplicated from signames.c */
 #if !defined (NSIG)
diff -Naur --exclude .git ../bash-v5.1-orig/support/signames.c ./support/signames.c
--- ../bash-v5.1-orig/support/signames.c	2025-01-02 18:28:47.242904223 -0500
+++ ./support/signames.c	2025-01-02 18:36:24.925759196 -0500
@@ -18,18 +18,12 @@
    along with Bash.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <config.h>
-
 #include <stdio.h>
 
 #include <sys/types.h>
 #include <signal.h>
+#include <stdlib.h>
 
-#if defined (HAVE_STDLIB_H)
-#  include <stdlib.h>
-#else
-#  include "ansi_stdlib.h"
-#endif /* HAVE_STDLIB_H */
 
 #if !defined (NSIG)
 #  define NSIG 64
