diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 8d323b3ee..f13eca078 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -3008,6 +3008,17 @@ netbsd*)
   hardcode_into_libs=yes
   ;;
 
+hades*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 newsos6)
   version_type=linux # correct to gnu/linux during the next big refactor
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
@@ -3674,6 +3685,10 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+hades*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 *-mlibc)
   lt_cv_deplibs_check_method=pass_all
   ;;
