diff -Naur ../zlib-v1.2.12-orig/configure ./configure
--- ../zlib-v1.2.12-orig/configure	2024-12-27 00:23:58.401605299 -0500
+++ ./configure	2024-12-27 00:24:41.413671385 -0500
@@ -218,7 +218,7 @@
     uname=`(uname -s || echo unknown) 2>/dev/null`
   fi
   case "$uname" in
-  Linux* | linux* | GNU | GNU/* | solaris*)
+  Linux* | linux* | GNU | GNU/* | solaris* | hades*)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"} ;;
   *BSD | *bsd* | DragonFly)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"}
