diff -Naur shadow-v4.8.1-orig/autogen.sh shadow-v4.8.1/autogen.sh
--- shadow-v4.8.1-orig/autogen.sh	2025-03-16 12:45:33.146028195 -0400
+++ shadow-v4.8.1/autogen.sh	2025-03-16 12:53:12.827648124 -0400
@@ -2,6 +2,8 @@
 
 autoreconf -v -f --install || exit 1
 
+exit 0
+
 ./configure \
 	CFLAGS="-O2 -Wall" \
 	--enable-man \
diff -Naur shadow-v4.8.1-orig/etc/login.defs shadow-v4.8.1/etc/login.defs
--- shadow-v4.8.1-orig/etc/login.defs	2025-03-16 12:45:33.146028195 -0400
+++ shadow-v4.8.1/etc/login.defs	2025-03-16 12:53:02.887679074 -0400
@@ -127,7 +127,7 @@
 #   Directory where mailboxes reside, _or_ name of file, relative to the
 #   home directory.  If you _do_ define both, MAIL_DIR takes precedence.
 #
-MAIL_DIR	/var/spool/mail
+MAIL_DIR	/var/mail
 #MAIL_FILE	.mail
 
 #
@@ -158,8 +158,8 @@
 # *REQUIRED*  The default PATH settings, for superuser and normal users.
 #
 # (they are minimal, add the rest in the shell startup files)
-ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
-ENV_PATH	PATH=/bin:/usr/bin
+ENV_SUPATH	PATH=/usr/sbin:/usr/bin
+ENV_PATH	PATH=/usr/bin
 
 #
 # Terminal permissions
@@ -188,7 +188,7 @@
 #
 # Prefix these values with "0" to get octal, "0x" to get hexadecimal.
 #
-ERASECHAR	0177
+ERASECHAR	010
 KILLCHAR	025
 #ULIMIT		2097152
 
@@ -332,7 +332,7 @@
 # Note: If you use PAM, it is recommended to use a value consistent with
 # the PAM modules configuration.
 #
-#ENCRYPT_METHOD DES
+#ENCRYPT_METHOD YESCRYPT
 
 #
 # Only works if ENCRYPT_METHOD is set to SHA256 or SHA512.
diff -Naur shadow-v4.8.1-orig/lib/encrypt.c shadow-v4.8.1/lib/encrypt.c
--- shadow-v4.8.1-orig/lib/encrypt.c	2025-03-16 12:45:33.150028081 -0400
+++ shadow-v4.8.1/lib/encrypt.c	2025-03-16 12:53:02.891679060 -0400
@@ -34,6 +34,7 @@
 
 #ident "$Id$"
 
+#include <crypt.h>
 #include <unistd.h>
 #include <stdio.h>
 
diff -Naur shadow-v4.8.1-orig/libmisc/user_busy.c shadow-v4.8.1/libmisc/user_busy.c
--- shadow-v4.8.1-orig/libmisc/user_busy.c	2025-03-16 12:45:33.154027966 -0400
+++ shadow-v4.8.1/libmisc/user_busy.c	2025-03-16 12:53:02.891679060 -0400
@@ -40,6 +40,7 @@
 #include <dirent.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <signal.h>
 #include "defines.h"
 #include "prototypes.h"
 #ifdef ENABLE_SUBIDS
diff -Naur shadow-v4.8.1-orig/libmisc/utmp.c shadow-v4.8.1/libmisc/utmp.c
--- shadow-v4.8.1-orig/libmisc/utmp.c	2025-03-16 12:45:33.154027966 -0400
+++ shadow-v4.8.1/libmisc/utmp.c	2025-03-16 12:53:02.891679060 -0400
@@ -47,6 +47,9 @@
 #include <netdb.h>
 #include <stdio.h>
 
+#include <fcntl.h>
+#include <netinet/in.h>
+
 #ident "$Id$"
 
 
diff -Naur shadow-v4.8.1-orig/Makefile.am shadow-v4.8.1/Makefile.am
--- shadow-v4.8.1-orig/Makefile.am	2025-03-16 12:45:33.146028195 -0400
+++ shadow-v4.8.1/Makefile.am	2025-03-16 12:53:02.891679060 -0400
@@ -2,5 +2,5 @@
 
 EXTRA_DIST = NEWS README TODO shadow.spec.in
 
-SUBDIRS = po man libmisc lib src \
+SUBDIRS = po libmisc lib src \
 	contrib doc etc
diff -Naur shadow-v4.8.1-orig/src/Makefile.am shadow-v4.8.1/src/Makefile.am
--- shadow-v4.8.1-orig/src/Makefile.am	2025-03-16 12:45:33.174027395 -0400
+++ shadow-v4.8.1/src/Makefile.am	2025-03-16 12:53:02.891679060 -0400
@@ -23,14 +23,12 @@
 # and installation would be much simpler (just two directories,
 # $prefix/bin and $prefix/sbin, no install-data hacks...)
 
-bin_PROGRAMS   = groups login
-sbin_PROGRAMS  = nologin
-ubin_PROGRAMS  = faillog lastlog chage chfn chsh expiry gpasswd newgrp passwd
+ubin_PROGRAMS  = faillog lastlog chage chfn chsh expiry gpasswd newgrp passwd login
 if ENABLE_SUBIDS
 ubin_PROGRAMS += newgidmap newuidmap
 endif
 if WITH_SU
-bin_PROGRAMS  += su
+ubin_PROGRAMS  += su
 endif
 usbin_PROGRAMS = \
 	chgpasswd \
@@ -44,6 +42,7 @@
 	grpunconv \
 	logoutd \
 	newusers \
+	nologin \
 	pwck \
 	pwconv \
 	pwunconv \
@@ -59,7 +58,7 @@
 suidbins       =
 suidubins      = chage chfn chsh expiry gpasswd newgrp
 if WITH_SU
-suidbins      += su
+suidubins      += su
 endif
 if !WITH_TCB
 suidubins += passwd
